.codecov.yml

```diff
@@ -0,0 +1,35 @@
+# https://docs.codecov.io/docs/commit-status
+coverage:
+  status:
+    # only measure diff changes, not unexpected project changes from flakiness
+    project: no
+    patch: yes
+      # default:
+        # https://docs.codecov.io/docs/commit-status#section-threshold
+        # TODO: consider setting instead of using existing coverage as baseline
+        # target: 
+        # https://docs.codecov.io/docs/commit-status#section-excluding-tests-example-
+        # TODO: should any paths be excluded from coverage metrics?
+        # paths:
+    # https://docs.codecov.io/docs/commit-status#section-changes-status
+    # TODO: enable after eliminating current unexpected coverage changes
+    changes: no
```

Comment:
wait why don't we want `changes` right now? what are the `"current unexpected coverage changes"`?

Reply:
`changes` would set a failing GitHub status for any coverage changes that aren't related to lines changed in the PR, such as all the changes in https://codecov.io/gh/hashicorp/consul/compare/be8fd29052b48410b72955c78356bcd4e48c89c4...92845f2a8227b88e09451841fd652d537e69f484/changes

One reason for current inconsistencies like ^ is our reliance on retries in tests, which for example may actually count the `return` line as covered if a test fails once then passes after retrying (and the error case wasn't explicitly tested otherwise).

```
if err != nil {
    return nil, err
}
```

I'd love to enable this eventually, but we'd want to eliminate current inconsistencies first to avoid frequently setting a failing status on most PRs that wouldn't be actionable.

Reply:
huh, i read the `changes` docs as saying, if you change code in part A of consul that affects the coverage of other code in part B, then show that code B coverage changed, even though code B didn't actually change itself.
the example they gave was removing a test in blah_test.go for function in blah.go would highlight blah.go as coverage being changed. that seems like something we would want?

The docs are super confusing and there seems to be overlap in a lot of their terminology. Did I just misunderstand what `changes` are?

https://docs.codecov.io/docs/commit-status#section-changes-status

Reply:
Nevermind! I understand what you meant now.